storage/libradb/src/lib.rs

```diff
@@ -127,32 +123,67 @@ impl LibraDB {
             TRANSACTION_ACCUMULATOR_CF_NAME,
             TRANSACTION_BY_ACCOUNT_CF_NAME,
             TRANSACTION_INFO_CF_NAME,
-        ];
+        ]
+    }
+
+    fn new_with_db(db: DB, prune_window: Option<u64>) -> Self {
+        let db = Arc::new(db);
+
+        LibraDB {
+            db: Arc::clone(&db),
+            event_store: EventStore::new(Arc::clone(&db)),
+            ledger_store: Arc::new(LedgerStore::new(Arc::clone(&db))),
+            state_store: Arc::new(StateStore::new(Arc::clone(&db))),
+            transaction_store: Arc::new(TransactionStore::new(Arc::clone(&db))),
+            system_store: SystemStore::new(Arc::clone(&db)),
+            pruner: prune_window.map(|n| Pruner::new(Arc::clone(&db), n)),
+        }
+    }
+
+    pub fn open<P: AsRef<Path> + Clone>(
+        db_root_path: P,
+        readonly: bool,
```

Comment:
should we make readonly always use "open as secondary ?"

Reply:
We'd need to update call sites to pass in the secondary path. (Using a tempdir is not always the right thing to do.)